src/modules/frei0r/frei0r_helper.c

```diff
@@ -104,23 +103,18 @@ int process_frei0r_item( mlt_service service, double position, double time, mlt_
 	int slice_count = mlt_properties_get(prop, "threads") ? mlt_properties_get_int(prop, "threads") : -1;
 	mlt_slices slices = NULL;
 
-	mlt_service_lock( service );
-
 	if (slice_count >= 0) {
-		slices = mlt_properties_get_data(prop, "mlt_slices", NULL);
-		if (!slices)
-			slices = mlt_slices_init(slice_count, SCHED_OTHER, sched_get_priority_max(SCHED_OTHER));
-		if (slices) {
-			mlt_properties_set_data(prop, "mlt_slices", slices, 0, (mlt_destructor) mlt_slices_close, NULL);
+		if ((slices = mlt_slices_get_global(mlt_policy_normal)))
```

Comment:
I might be missing something here. But it seems to me that this will never be true because slices is initialized to NULL above.